de.dlr.sc.virsat.model.extension.fdir/src/de/dlr/sc/virsat/model/extension/fdir/util/RecoveryAutomatonHelper.java

```diff
@@ -132,6 +132,38 @@ public void assignInputs(FaultEventTransition transition, FaultTreeNode... fault
 		return transition.getRecoveryActions();
 	}
 	
+	/**
+	 * Determine if two states in a recovery automaton are connected by a transition
+	 * @param ra the recovery automaton
+	 * @param from the state where the transition is from
+	 * @param to the state where the transition is to
+	 * @return true if there exists such a transition, false otherwise
+	 */
+	public boolean isConnected(RecoveryAutomaton ra, State from, State to) {
+		for (Transition transition : ra.getTransitions()) {
+			if (transition.getFrom().getName().equals(from.getName()) && transition.getTo().getName().equals(to.getName())) {

```

Comment:
Do we need to use the names? I would expected the method to compare the actual states ie `transition.getFrom().equals(from)`